lib/elastix/http.ex

```diff
@@ -18,20 +18,14 @@ defmodule Elastix.HTTP do
     full_url = process_url(to_string(query_url))
     body = process_request_body(body)
 
-    username = Elastix.config(:username)
-    password = Elastix.config(:password)
-
-    content_headers = headers
-    |> Keyword.put_new(:"Content-Type", "application/json; charset=UTF-8")
-
-    full_headers = if Elastix.config(:shield) do
-      Keyword.put(content_headers, :"Authorization", "Basic " <> Base.encode64("#{username}:#{password}"))
-    else
-      content_headers
-    end
+    full_headers = headers
+                   |> add_content_type_header
+                   |> add_shield_header
+                   |> add_custom_headers(method, full_url, body)
 
     options = Keyword.merge(default_httpoison_options(), options)
-    HTTPoison.Base.request(
+    {m, f, _a} = Elastix.config(:test_request_mfa) || {HTTPoison.Base, :request, []}
```

Comment:
This could be used to override HTTP lib, can we rename this `test_request_mfa` atom?